src/app/loadout/getMaximumPotentiallyAvailableArmorSlotEnergyCapacities.ts

```diff
@@ -0,0 +1,120 @@
+import { PluggableInventoryItemDefinition } from 'app/inventory/item-types';
+import { MAX_ARMOR_ENERGY_CAPACITY } from 'app/search/d2-known-values';
+import { errorLog } from 'app/utils/log';
+import { BucketHashes } from 'data/d2/generated-enums';
+import _ from 'lodash';
+import memoizeOne from 'memoize-one';
+import { slotSpecificPlugCategoryHashes } from './known-values';
+import { categorizeArmorMods, getModCost } from './mod-assignment-utils';
+import { generatePermutationsOfFive } from './mod-permutations';
+import {
+  ArmorBucketHashes,
+  OrderedArmorBucketHashes,
+  getBucketHashFromPlugCategoryHash,
+} from './mod-utils';
+
+export type ArmorBucketHashToNumberRecord = Record<ArmorBucketHashes, number>;
+
+export const getDefaultSlotSpecificBucketHashToNumberRecord =
+  (): ArmorBucketHashToNumberRecord => ({
+    [BucketHashes.Helmet]: 0,
+    [BucketHashes.Gauntlets]: 0,
+    [BucketHashes.ChestArmor]: 0,
+    [BucketHashes.LegArmor]: 0,
+    [BucketHashes.ClassArmor]: 0,
+  });
+
+export function sumParallelArrays(arrays: number[][]): number[] {
+  // If all arrays are not the same length, throw an error
+  if (!arrays.every((arr) => arr.length === arrays[0].length)) {
+    errorLog('loadouts', 'All arrays must be the same length');
+    return [];
+  }
+  // Use lodash's zipWith function to combine the arrays element-wise and sum them
+  const result = _.zipWith(...arrays, (...values: number[]) => _.sum(values));
+
+  return result;
+}
+
+// Pad an array to length 5 with 0s
+export const padCostArray = (arr: number[]) => Array.from({ ...arr, length: 5 }, (v) => v ?? 0);
```

Comment:
Maybe these should be in `utils/collections` (perhaps with a more generic name)?

`padCostArray` doesn't only pad, it discards any excess elements!

Reply:
Oh shoot I did not realize it was doing that. I'll find a better way to do that and I can move these functions.